File: Frontend-BAP/screens/CreateProcurementScreen.js
import {
  View,
  Text,
  TouchableOpacity,
  Image,
  TextInput,
  ScrollView,
} from 'react-native';
import { useState } from 'react';
import backArrow from '../assets/left-arrow.png';
import { CameraIcon } from 'react-native-heroicons/outline';
import tw from 'tailwind-react-native-classnames';
import { useNavigation } from '@react-navigation/native';

const CreateProcurementScreen = () => {
  const navigation = useNavigation();

  // State for input fields
  const [farmerName, setFarmerName] = useState('');
  const [farmerPhone, setFarmerPhone] = useState('');
  const [cropName, setCropName] = useState('');
  const [varietyName, setVarietyName] = useState('');
  const [deliveryPersonName, setDeliveryPersonName] = useState('');
  const [deliveryPersonMobile, setDeliveryPersonMobile] = useState('');
  const [tractorNo, setTractorNo] = useState('');
  const [sameAsFarmer, setSameAsFarmer] = useState(false);

  return (
    <View className='flex-1 relative h-full bg-white'>
      <View
        style={tw`flex-row justify-start items-center p-2`}
        className='bg-[#B2FFB7]'
      >
        <TouchableOpacity onPress={() => navigation.navigate('Procurements')}>
          <Image source={backArrow} style={tw`w-6 h-6 p-1`} />
        </TouchableOpacity>
        <Text style={tw`text-xl font-semibold`}>Add New Procurement</Text>
      </View>

      <ScrollView className='flex-col space-y-2 p-4'>
        <Text className='font-bold'>Farmer Details</Text>

        <Text>Farmer Name</Text>
        <TextInput
          className='border-2 border-[#D9D9D9] rounded-lg p-2'
          value={farmerName}
          onChangeText={setFarmerName}
        />

        <Text>Phone No.</Text>
        <TextInput
          className='border-2 border-[#D9D9D9] rounded-lg p-2'
          value={farmerPhone}
          onChangeText={setFarmerPhone}
        />

        <Text className='font-bold'>Crop Details</Text>
        <TextInput
          className='border-2 border-[#D9D9D9] rounded-lg p-2'
          placeholder='Crop Name'
          value={cropName}
          onChangeText={setCropName}
        />
        <TextInput
          className='border-2 border-[#D9D9D9] rounded-lg p-2'
          placeholder='Variety Name'
          value={varietyName}
          onChangeText={setVarietyName}
        />
        <Text className='font-semibold'>Upload or Click Crop Photo</Text>
        <TouchableOpacity className='flex  justify-center items-center h-[40%] bg-[#D9D9D9] rounded-lg '>
          <CameraIcon />
        </TouchableOpacity>

        <Text className='font-bold'>Delivery Person Details</Text>
        {!sameAsFarmer && (
          <View className='flex-col h-full space-y-2'>
            <TextInput
              className='border-2 border-[#D9D9D9] rounded-lg p-2'
              placeholder='Name'
              value={deliveryPersonName}
              onChangeText={setDeliveryPersonName}
            />
            <TextInput
              className='border-2 border-[#D9D9D9] rounded-lg p-2'
              placeholder='Mobile No.'
              value={deliveryPersonMobile}
              onChangeText={setDeliveryPersonMobile}
            />
            <TextInput
              className='border-2 border-[#D9D9D9] rounded-lg p-2'
              placeholder='Tractor No.'
              value={tractorNo}
              onChangeText={setTractorNo}
            />

            <Text className='font-semibold'>Upload or Click Tractor Photo</Text>
            <TouchableOpacity className='flex justify-center items-center h-[40%] bg-[#D9D9D9] rounded-lg'>
              <CameraIcon />
            </TouchableOpacity>
          </View>
        )}
      </ScrollView>

      <View className='bg-[#ecfcf4] w-full absolute bottom-0 justify-center items-center py-2 px-4'>
        <TouchableOpacity className='bg-[#2B9846] justify-center items-center w-full p-2 rounded-lg'>
          <Text className='text-white'>Save</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default CreateProcurementScreen;

File: demo/lib/screens/create_procurement_screen.dart
import 'dart:io';

import 'package:flutter/material.dart';
// ignore: depend_on_referenced_packages
import 'package:image_picker/image_picker.dart';

class CreateProcurementScreen extends StatefulWidget {
  const CreateProcurementScreen({super.key});

  @override
  State<CreateProcurementScreen> createState() =>
      _CreateProcurementScreenState();
}

class _CreateProcurementScreenState extends State<CreateProcurementScreen> {
  bool sameAsFarmer = false;

  // Controllers for input fields
  final TextEditingController farmerNameCtrl = TextEditingController();
  final TextEditingController farmerPhoneCtrl = TextEditingController();
  final TextEditingController cropNameCtrl = TextEditingController();
  final TextEditingController varietyNameCtrl = TextEditingController();
  final TextEditingController deliveryNameCtrl = TextEditingController();
  final TextEditingController deliveryPhoneCtrl = TextEditingController();
  final TextEditingController tractorNoCtrl = TextEditingController();

  File? cropImage;
  File? tractorImage;

  final ImagePicker picker = ImagePicker();

  Future<void> pickImage(bool isCrop) async {
    final pickedFile =
        await picker.pickImage(source: ImageSource.gallery); // or camera
    if (pickedFile != null) {
      setState(() {
        if (isCrop) {
          cropImage = File(pickedFile.path);
        } else {
          tractorImage = File(pickedFile.path);
        }
      });
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
      body: Stack(
        children: [
          Column(
            children: [
              // Header
              Container(
                color: const Color(0xFFB2FFB7),
                padding: const EdgeInsets.all(8),
                child: Row(
                  children: [
                    IconButton(
                      icon: const Icon(Icons.arrow_back, size: 24),
                      onPressed: () {
                        Navigator.pop(context);
                      },
                    ),
                    const Text(
                      "Add New Procurement",
                      style:
                          TextStyle(fontSize: 18, fontWeight: FontWeight.w600),
                    )
                  ],
                ),
              ),

              // Scrollable form
              Expanded(
                child: SingleChildScrollView(
                  padding: const EdgeInsets.all(12),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      const Text("Farmer Details",
                          style: TextStyle(
                              fontSize: 16, fontWeight: FontWeight.bold)),
                      inputField(controller: farmerNameCtrl, label: "Farmer Name"),
                      inputField(controller: farmerPhoneCtrl, label: "Phone No."),

                      const SizedBox(height: 12),
                      const Text("Crop Details",
                          style: TextStyle(
                              fontSize: 16, fontWeight: FontWeight.bold)),
                      inputField(controller: cropNameCtrl, label: "Crop Name"),
                      inputField(controller: varietyNameCtrl, label: "Variety Name"),

                      const SizedBox(height: 8),
                      const Text("Upload or Click Crop Photo",
                          style: TextStyle(fontWeight: FontWeight.w600)),
                      const SizedBox(height: 6),
                      GestureDetector(
                        onTap: () => pickImage(true),
                        child: Container(
                          height: 180,
                          width: double.infinity,
                          decoration: BoxDecoration(
                            color: const Color(0xFFD9D9D9),
                            borderRadius: BorderRadius.circular(10),
                          ),
                          child: cropImage == null
                              ? const Center(child: Icon(Icons.camera_alt, size: 36))
                              : Image.file(cropImage!, fit: BoxFit.cover),
                        ),
                      ),

                      const SizedBox(height: 12),
                      const Text("Delivery Person Details",
                          style: TextStyle(
                              fontSize: 16, fontWeight: FontWeight.bold)),

                      if (!sameAsFarmer) ...[
                        inputField(controller: deliveryNameCtrl, label: "Name"),
                        inputField(controller: deliveryPhoneCtrl, label: "Mobile No."),
                        inputField(controller: tractorNoCtrl, label: "Tractor No."),

                        const SizedBox(height: 8),
                        const Text("Upload or Click Tractor Photo",
                            style: TextStyle(fontWeight: FontWeight.w600)),
                        const SizedBox(height: 6),
                        GestureDetector(
                          onTap: () => pickImage(false),
                          child: Container(
                            height: 180,
                            width: double.infinity,
                            decoration: BoxDecoration(
                              color: const Color(0xFFD9D9D9),
                              borderRadius: BorderRadius.circular(10),
                            ),
                            child: tractorImage == null
                                ? const Center(child: Icon(Icons.camera_alt, size: 36))
                                : Image.file(tractorImage!, fit: BoxFit.cover),
                          ),
                        ),
                      ],
                      const SizedBox(height: 80),
                    ],
                  ),
                ),
              ),
            ],
          ),

          // Save Button
          Positioned(
            bottom: 0,
            left: 0,
            right: 0,
            child: Container(
              color: const Color(0xFFecfcf4),
              padding: const EdgeInsets.all(12),
              child: ElevatedButton(
                style: ElevatedButton.styleFrom(
                  backgroundColor: const Color(0xFF2B9846),
                  padding: const EdgeInsets.symmetric(vertical: 14),
                  shape: RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(8),
                  ),
                ),
                onPressed: () {
                  final data = {
                    "farmerName": farmerNameCtrl.text,
                    "farmerPhone": farmerPhoneCtrl.text,
                    "cropName": cropNameCtrl.text,
                    "varietyName": varietyNameCtrl.text,
                    "deliveryName": deliveryNameCtrl.text,
                    "deliveryPhone": deliveryPhoneCtrl.text,
                    "tractorNo": tractorNoCtrl.text,
                    "cropImage": cropImage?.path,
                    "tractorImage": tractorImage?.path,
                  };
                  Navigator.pop(context, data); // send back to ProcurementsScreen
                },
                child: const Text(
                  "Save",
                  style: TextStyle(
                      color: Colors.white,
                      fontWeight: FontWeight.w600,
                      fontSize: 16),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }

  Widget inputField(
      {required TextEditingController controller, String label = ""}) {
    return Container(
      margin: const EdgeInsets.symmetric(vertical: 6),
      child: TextField(
        controller: controller,
        decoration: InputDecoration(
          labelText: label,
          border: OutlineInputBorder(
            borderRadius: BorderRadius.circular(8),
            borderSide: const BorderSide(color: Color(0xFFD9D9D9), width: 2),
          ),
        ),
      ),
    );
  }
}
